### ISSUE TITLE
Failed to call service

### ISSUE THREAD

**Issue by brianhanifin** [custom-components/config_check#1]
## Version of the custom_component
0.0.1 (according to the custom updater/tracker card)

## Describe the bug
I get a "Failed to call service config_check/run" error when attempting to call the service from the Services dev page and when attempting to call it from a Lovelace call-service call on an entities card.

## Configuration
* I include a reference to [`config_check:` in an included config package](https://github.com/brianhanifin/Home-Assistant-Config/blob/master/packages/config_check.yaml).
* In addition to attempting to call it from the dev-service page I included a call to it [n this Lovelace file](https://github.com/brianhanifin/Home-Assistant-Config/blob/master/lovelace/views/tools.yaml#L81).

I am running Home Assistant in a docker container. I have several other working custom components.

## Log

```text
Error handling message: {'type': 'call_service', 'domain': 'config_check', 'service': 'run', 'service_data': {}, 'id': 28}
Traceback (most recent call last):
  File "/usr/src/app/homeassistant/components/websocket_api/decorators.py", line 17, in _handle_async_response
    await func(hass, connection, msg)
  File "/usr/src/app/homeassistant/components/websocket_api/commands.py", line 147, in handle_call_service
    connection.context(msg))
  File "/usr/src/app/homeassistant/core.py", line 1133, in async_call
    self._execute_service(handler, service_call))
  File "/usr/src/app/homeassistant/core.py", line 1155, in _execute_service
    await handler.func(service_call)
  File "/config/custom_components/config_check/__init__.py", line 30, in run_check_service
    result = await run_check(str(hass.config.path()))
  File "/config/custom_components/config_check/__init__.py", line 41, in run_check
    stdin=PIPE, stdout=PIPE, stderr=PIPE)
  File "/usr/local/lib/python3.7/subprocess.py", line 775, in __init__
    restore_signals, start_new_session)
  File "/usr/local/lib/python3.7/subprocess.py", line 1522, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'hass': 'hass'
```

**Comment by brianhanifin** [comment]
I was able to get a useful response at the docker command line with the following command.
```
python -m homeassistant --config /config --script check_config --i | grep -A 1 ERROR
```

I purposefully broke a config line and it returned the following:
```
ERROR:homeassistant.util.yaml:mapping values are not allowed here
  in "/config/packages/config_check.yaml", line 10, column 6
```

**Comment by ludeeus** [comment]
Please provide debug logs.
`custom_components.config_check: debug` for [logger](https://www.home-assistant.io/components/logger/) component

**Comment by ludeeus** [comment]
Don't bother adding that, 0.0.2 will fix it.

**Comment by brianhanifin** [comment]
Here is the 0.0.2 debug log.

```
2019-03-01 06:53:37 DEBUG (MainThread) [custom_components.config_check] Could not find
homeassistant - [Errno 2] No such file or directory: 'homeassistant': 'homeassistant'

2019-03-01 06:53:37 CRITICAL (MainThread) [custom_components.config_check] Could not find
hass/homeassistant
```

For reference, here is the path and `homeassistant` python script directory structure for the [official homeassistant/home-assistant docker image](https://hub.docker.com/r/homeassistant/home-assistant). In this image this is app's path, while the configuration is stored in the path `/config`.
```
root@server:/usr/src/app/homeassistant# ls -l
total 228
-rw-r--r-- 1 root root    36 Feb 27 16:55 __init__.py
-rw-r--r-- 1 root root 12576 Feb 27 16:55 __main__.py
drwxr-xr-x 2 root root  4096 Feb 28 09:30 __pycache__
drwxr-xr-x 1 root root  4096 Feb 28 09:29 auth
-rw-r--r-- 1 root root 14013 Feb 27 16:55 bootstrap.py
drwxr-xr-x 1 root root  4096 Feb 28 09:30 components
-rw-r--r-- 1 root root 29708 Feb 27 16:55 config.py
-rw-r--r-- 1 root root 19382 Feb 27 16:55 config_entries.py
-rw-r--r-- 1 root root 12698 Feb 27 16:55 const.py
-rw-r--r-- 1 root root 43842 Feb 27 16:55 core.py
-rw-r--r-- 1 root root  5840 Feb 27 16:55 data_entry_flow.py
-rw-r--r-- 1 root root  2443 Feb 27 16:55 exceptions.py
drwxr-xr-x 1 root root  4096 Feb 28 09:30 helpers
-rw-r--r-- 1 root root  9803 Feb 27 16:55 loader.py
-rw-r--r-- 1 root root  2334 Feb 27 16:55 monkey_patch.py
-rw-r--r-- 1 root root   726 Feb 27 16:55 package_constraints.txt
-rw-r--r-- 1 root root  3593 Feb 27 16:55 requirements.py
drwxr-xr-x 1 root root  4096 Feb 28 12:32 scripts
-rw-r--r-- 1 root root  9350 Feb 27 16:55 setup.py
drwxr-xr-x 1 root root  4096 Feb 28 09:30 util
```

```
root@server:/config# ls -l
total 400
drwxr-xr-x  2 1000 1000  4096 Feb 11 11:02 README-images
-rwxr--r--  1 1000 1000 12540 Feb 11 11:09 README.md
drwxr-xr-x  5 1000 1000  4096 Jan 10 21:52 appdaemon
drwxr-xr-x 13 1000 1000  4096 Feb 21 14:24 automations
drwxr-xr-x 31 1000 1000  4096 Feb 21 12:42 components
-rwxr-xr-x  1 1000 1000  1190 Feb 18 18:24 configuration.yaml
drwxr-xr-x  9 1000 1000  4096 Feb 27 19:49 custom_components
drwxr-xr-x  5 1000 1000  4096 Jan 10 21:55 customize
drwxr-xr-x  3 1000 1000  4096 Jan 10 21:52 deps
-rw-r--r--  1 root root 27225 Mar  1 07:01 home-assistant.log
drwxr-xr-x  2 1000 1000  4096 Jan 10 21:52 json_data
-rwxr-xr-x  1 1000 1000 14444 Feb 22 13:47 known_devices.yaml
drwxr-xr-x  4 1000 1000  4096 Jan 10 21:55 lovelace
drwxr-xr-x  2 1000 1000  4096 Mar  1 06:42 packages
drwxr-xr-x  2 1000 1000  4096 Feb 12 16:52 python_scripts
drwxr-xr-x  4 1000 1000  4096 Jan 10 21:52 scenes
drwxr-xr-x 14 1000 1000  4096 Feb 18 13:24 scripts
-rwxr-xr-x  1 1000 1000  1854 Feb 23 08:12 secrets.yaml
drwxr-xr-x  2 1000 1000  4096 Mar  1 06:38 shell_commands
drwxr-xr-x  3 1000 1000  4096 Jan 10 21:52 templates
drwxr-xr-x  2 1000 1000  4096 Jan 10 21:52 themes
-rwxr-xr-x  1 1000 1000  1234 Feb 23 11:14 travis_secrets.yaml
-rwxr-xr-x  1 1000 1000     0 Feb 12 18:21 ui-configuration.yaml
-rwxr-xr-x  1 1000 1000   761 Feb 15 22:31 ui-lovelace.yaml
drwxr-xr-x  8 1000 1000  4096 Feb 18 11:23 www
```

**Comment by ludeeus** [comment]
If this still does not work the location of hass (cli) are not in PATH

**Comment by brianhanifin** [comment]
OK, I agree that this isn't going to work then. Here is the output of the $PATH. And altering it wouldn't do any good because it gets overwritten when being upgraded. I'm going to see if I can get it working as a command_line sensor. Thank you for looking into it.
```
/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

**Comment by ludeeus** [comment]
I don't give up that easy ;)
try 0.0.3

**Comment by brianhanifin** [comment]
Bingo! I have always wanted to do this. Ever since I left the Raspberry Pi behind I have had to swap between the config tab and my Tools tab to check, then click restart. Docker users cannot simply use the built in restart/reload commands on the configuration check page. The process is tedious enough that I often skip checking the config.

Is the result stored anywhere, perhaps in a sensor? If it did then I could setup a script to only reboot if the config check was a success. Even without that your component has made a huge quality of life improvement for me. Thank you.

Here is screenshot proof! (I intentionally created a validation problem to test this.)

<img width="813" alt="screen shot 2019-03-01 at 9 07 34 am" src="https://user-images.githubusercontent.com/57336/53654216-07349b00-3c02-11e9-9cd1-618fed40f557.png">

**Comment by ludeeus** [comment]
Awesome!
The state is currently not stored, but if you give me what you use to restart I can probably add a second service that include restart on success

**Comment by brianhanifin** [comment]
TLDR: I call a shell command called `shell_command.restart_hass`. So if you added a parameter that I could set (perhaps called `on_success_service_call`) anyone could configure a script or shell_command to run on success.

Details: I am using HADockermon to restart the Home Assistant Docker with a `shell_command` that pings a REST API.
```
curl http://10.0.0.56:8126/container/homeassistant/restart
```

**Comment by brianhanifin** [comment]
This problem I had was resolved heroically by @ludeeus! I felt like it would be more productive to move the feature request into a new issue (#2) and close this one.
